### PR TITLE
ci(ocaml5): get rid of ocaml 4.08 and add ocaml 5

### PR DESCRIPTION
**Pull request by gsebil08**
## Depends
- [x] #28

### CODE REVIEW

File: .github/workflows/esy-test.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Pollinate tests
```
Comment (gsebil08) [inline .github/workflows/esy-test.yml:1]: Added the `esy` version, since we are using `esy x dune test`

**Comment by rosalogia** [comment]
I don't know much about CI, so this looks okay to me, but it's probably best if we have @ulrikstrid weigh in if you're unsure about some things.

File: .github/workflows/esy-test.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - main
+  schedule:
+    # Prime the caches every Monday
+    - cron: 0 1 * * MON
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (gsebil08) [inline .github/workflows/esy-test.yml:16]: @ulrikstrid I think it exists a better way to only use one `os`, right?

Comment (ulrikstrid) [reply .github/workflows/esy-test.yml:16]: Yeah, just remove the matrix all together, the matrix is only useful if you want to test multiple combinations

File: .github/workflows/opam-build.yml [hunk 3]
```diff
@@ -17,21 +17,19 @@ jobs:
...
 
-      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+      - name: Use OCaml 5
         uses: ocaml/setup-ocaml@v2
-        with:
-          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam repo add beta https://github.com/ocaml/ocaml-beta-repository.git
+
+      - run: opam switch reinstall . --no-install --packages ocaml.5.00.0
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply .github/workflows/opam-build.yml:32]: I can't come up with a better solution so this is probably fine

**Comment by gsebil08** [comment]
@rosalogia @ulrikstrid should I keep the 4.13 also?
I think it is useless since I added the `esy` workflow, which will test the `4.13` since we have it within `package.json`
```json
  "dependencies": {
    "ocaml": "4.13.x"
    [...]
  }
```
What do you think?

**Comment by ulrikstrid** [comment]
@gsebil08 Since the goal is to use this via deku I would rather have ocaml 5 in esy and then test other versions in opam

**Review by rosalogia — APPROVED** [review]

**Pull request merged**